File: ts-node/src/web/application.ts
import * as http from "http";
import {Router} from "./router";
import {Context} from "./context";

export class Application extends Router {
    server: http.Server;
    errorHandler: (err: Error, ctx: Context) => Promise<void>;
    constructor() {
        super("");
  
        this.errorHandler = async (err, ctx) => {
            ctx.statusCode = 500;
            ctx.send(err.stack ?? 'server error');
        }
        // 查询中间执行函数
        this.server = http.createServer((req, res) => {
            const ctx = new Context(req, res);
            ctx.errorHandler = this.errorHandler;
            ctx.timeout(10 * 1000);
            this.find(ctx);
        });
    }

    listen(addr: string) {
        this.server.listen(addr);
    }
}

File: dart/lib/src/application.dart
import 'dart:io';

import 'context.dart';
import 'router.dart';

class Application extends Router {
  HttpServer server;
  Application() : super("ANY", "") {

  }

  listen(String addr, int port) async {
    this.server = await HttpServer.bind(addr, port);
    this.server.listen((req) {
      var ctx = new Context(req, req.response);
      this.find(ctx);
    });
  }
}